platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -131,6 +153,182 @@ null_crypto_routine(odp_packet_t pkt ODP_UNUSED,
 	return ODP_CRYPTO_ALG_ERR_NONE;
 }
 
+static
+odp_crypto_alg_err_t aes_gcm_encrypt(odp_packet_t pkt,
+				     const odp_crypto_packet_op_param_t *param,
+				     odp_crypto_generic_session_t *session)
+{
+	armv8_cipher_state_t cs = {
+		.counter = {
+			.d = {0, 0}
+		}
+	};
+	uint8_t *iv_ptr;
+	uint64_t plaintext_bit_length = param->cipher_range.length * 8;
+	uint64_t iv_bit_length = session->p.cipher_iv.length * 8;
+	uint64_t aad_bit_length = session->p.auth_aad_len * 8;
+	uint32_t in_pos = param->cipher_range.offset;
+	uint32_t in_len = param->cipher_range.length;
+	int ret = 0;
+
+	/* Fail early if cipher_range is too large */
+	if (in_len > _ARM_CRYPTO_MAX_DATA_LENGTH) {
+		ODP_DBG("ARM Crypto: Packet size too large for requested operation\n");
+		ret = -1;
+		goto err;
+	}
+
+	if (param->cipher_iv_ptr)
+		iv_ptr = param->cipher_iv_ptr;
+	else if (session->p.cipher_iv.data)
+		iv_ptr = session->cipher.iv_data;
+	else
+		return ODP_CRYPTO_ALG_ERR_IV_INVALID;
+
+	cs.constants = &session->cc;
+
+	if (armv8_aes_gcm_set_counter(iv_ptr, iv_bit_length, &cs) != 0) {
+		ODP_DBG("ARM Crypto: Failure while setting nonce\n");
+		ret = -1;
+		goto err;
+	}
+
+	uint32_t seg_len = 0;
+	uint8_t *data = odp_packet_offset(pkt, in_pos, &seg_len, NULL);
+
+	if (in_len > seg_len) {
+		/* Packet is segmented within the cipher range. Copy the cipher
+		 * range to a contiguous buffer. */
+		odp_packet_copy_to_mem(pkt, in_pos, in_len, local.buffer);
+
+		data = local.buffer;
+	}
+
+	if (armv8_enc_aes_gcm_from_state(&cs,
+					 param->aad_ptr, aad_bit_length,
+					 data, plaintext_bit_length,
+					 local.buffer,
+					 local.digest) != 0) {
+		ODP_DBG("ARM Crypto: AES GCM Encoding failed\n");
+		ret = -1;
+		goto err;
+	}
+
+	odp_packet_copy_from_mem(pkt, param->cipher_range.offset,
+				 param->cipher_range.length, local.buffer);
+
+	/* Clear the hash bytes */
+	odp_packet_copy_from_mem(pkt, param->hash_result_offset,
```

Comment:
This added line does not accomplish anything. The next line writes over the very same bytes. You probably intended to add this before the encryption call, but actually even that probably should not be done (see PR 1435 for an API change to make the current linux-gen behaviour ok).

Reply:
I had added this code because in the last review, you had given a comment: https://github.com/OpenDataPlane/odp/pull/1253#discussion_r749344579, which said that the ODP API specifies that the hash result bytes should be cleared before the hash calculation is done. I had seen in some other files (udp I think) that ZEROS array was used with `odp_packet_copy_from_mem` to clear bytes within a specified range in a packet so used the same methodology. 
I had intended to add this before the encryption call but incorrectly placed this code here. Since the spec has changed and this behavior is no longer needed, I will remove this code. Thanks!

Reply:
The spec has not changed yet, but this is probably ok since it is the same as in the openssl_crypto now.